packages/melos/lib/src/commands/bootstrap.dart

```diff
@@ -343,7 +351,9 @@ mixin _BootstrapMixin on _CleanMixin {
     // dependencies that have a different version specified in the workspace.
     final dependenciesToUpdate = workspaceDependencies.entries.where((entry) {
       if (!packageDependencies.containsKey(entry.key)) return false;
-      if (packageDependencies[entry.key] == entry.value) return false;
+      if (_areDependenciesEqual(packageDependencies[entry.key], entry.value)) {
```

Comment:
Can you add a `TODO` here that we should remove this when we change or upgrade pubspec dependency?

Reply:
Done.